src/vsr/replica.zig

```diff
@@ -2469,11 +2489,6 @@ pub fn ReplicaType(
                 stage.trailers.get(trailer).offset,
                 target_size,
             });
-
-            if (stage.done()) {
-                assert(progress == .complete);
-                self.sync_requesting_trailers_callback();
-            }
```

Comment:
`on_sync_checkpoint()` just checks `if (stage.done())` before accepting the "checkpoint state".

Since the other trailer steps are going to go away soon the performance impact of potentially ignoring a `sync_checkpoint` that arrives before the trailers won't be an issue, and it keeps things simpler.